Automattic/jetpack
# Switch to useModuleStatus hook instead of directly calling data store

Fixes https://github.com/Automattic/jetpack/issues/34603

## Proposed changes:

After we enhance `useModuleStatus` using data store there's no point calling it directly and we can utilize hook instead.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Clone the branch locally.
* Build `packages/forms` and `plugins/jetpack`
* Navigate to `wp-admin/admin.php?page=jetpack_modules` and disable `Contact Form` module
* Create a new post and add Contact Form to the page
* No AI assistant should be suggested when module not activated
* Activate the module and observe the behaviour. Everything should work as before. You should be able to enable the module and see AI assistant after.
